#! /bin/sh -e

# DP: Fix profiled build; don't use Python/thread.gc*, gcc complains

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

--- Makefile.pre.in.orig	2009-02-11 12:50:56.000000000 +0100
+++ Makefile.pre.in	2009-02-11 12:51:30.000000000 +0100
@@ -370,9 +370,11 @@
 	$(MAKE) all CFLAGS="$(CFLAGS) -fprofile-generate" LIBS="$(LIBS) -lgcov"
 
 run_profile_task:
-	./$(BUILDPYTHON) $(PROFILE_TASK)
+	-./$(BUILDPYTHON) $(PROFILE_TASK)
 
 build_all_use_profile:
+	: # FIXME: gcc error
+	rm -f Python/thread.gc*
 	$(MAKE) all CFLAGS="$(CFLAGS) -fprofile-use"
 
 coverage:
